go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -16,154 +18,198 @@ import (
 	"k8s.io/klog/v2"
 )
 
-type gatewayInfoList []*gatewayInfo
+func insertSet(s1, s2 sets.Set[string]) {
+	for insertItem := range s2 {
+		s1.Insert(insertItem)
+	}
+}
+
+// gatewayInfoList stores a list of gatewayInfo with unique ips
+type gatewayInfoList struct {
+	elems []*gatewayInfo
```

Comment:
This was done to make sure there are no 2 etries with the same ip, so the only way to add elements should be via `InsertOverwrite` that has a tricky logic of removing duplicated ips

Reply:
sorry I still dont understand. Why couldn't you do:
```
type gatewayInfoList []*gatewayInfo
```
and still use InsertOverwrite?

Reply:
so I tried to make sure no one will update elements directly (which is still possible), and I thought that `append(allGWIPs.elems, p.staticGateways...)`  is a bit more obviously wrong than
`append(allGWIPs, p.staticGateways...)`.
But I can just move it to a separate package, so that there is not way to do that at all.

Reply:
yeah I guess if your intention is to protect someone from accessing the slice, we should move to another package